File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int cur = nums[0];
        int count = 1;
        for (int i=1; i<n; i++){
            if (nums[i] == cur)
                count++;
            else if (--count == 0){
                cur = nums[i];
                count = 1;
            }
        }
        return cur;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cnt = 1
        cur = nums[0]
        for num in nums[1:]:
            if num == cur:
                cnt += 1
            else:
                cnt -= 1
                if cnt < 0:
                    cnt = 1
                    cur = num
        return cur